airflow/contrib/hooks/telegram_hook.py

```diff
@@ -0,0 +1,117 @@
+                        logger.warning(msg)
+                    else:
+                        print(msg)
+                    time.sleep(mdelay)
+                    mtries -= 1
+                    mdelay *= backoff
+            return f(*args, **kwargs)
+
+        return f_retry  # true decorator
+
+    return deco_retry
+
+
+class TelegramHook(BaseHook):
+    """
+       Interact with Telegram, using python-telegram-bot library.
+    """
+
+    def __init__(self, telegram_conn_id=None, chat_id=None):
+        """
+        Takes both telegram bot API token directly and connection that has telegram bot API token.
+        If both supplied, telegram API token will be used.
+        
+        :param telegram_conn_id: connection that has telegram API token in the password field
+        :type telegram_conn_id: str
+        :param chat_id: Telegram public or private channel id (optional).
+        Check https://stackoverflow.com/a/33862907 to see how you can obtain chat_id for private
+        channel
+        :type chat_id: str
+        """
+        self.token = self.__get_token(telegram_conn_id)
+        self.chat_id = self.__get_chat_id(chat_id, telegram_conn_id)
+        self.connection = self.get_conn()
+
+    def get_conn(self):
+        return telegram.Bot(token=self.token)
+
+    def __get_token(self, telegram_conn_id):
+        if telegram_conn_id is not None:
+            conn = self.get_connection(telegram_conn_id)
+
+            if not conn.password:
+                raise AirflowException("Missing token(password) in Telegram connection")
+            return conn.password
+        else:
+            raise AirflowException(
+                "Cannot get token: " "No valid Telegram connection supplied."
+            )
+
+    def __get_chat_id(self, chat_id, telegram_conn_id):
+        if chat_id is not None:
+            return chat_id
+        elif telegram_conn_id is not None:
+            conn = self.get_connection(telegram_conn_id)
+
+            if not getattr(conn, "host", None):
+                raise AirflowException("Missing chat_id (host) in Telegram connection")
+            return conn.host
+        else:
+            raise AirflowException(
+                "Cannot get chat_id: " "No valid chat_id nor telegram_conn_id supplied."
+            )
+
+    @retry(exceptions=telegram.error.TelegramError, tries=5, delay=0)
+    def call(self, method, api_params):
+        """
+        Send a message to a telegram channel
+
+        :param method: not used
+        :type method: str
+        :param api_params: params for telegram_instance.send_message. You can use it also to override chat_id
+        :type api_params: dict
+        """
+
+        params = {
+            "chat_id": self.chat_id,
+            "parse_mode": telegram.ParseMode.HTML,
+            "disable_web_page_preview": True,
+        }
+        params.update(api_params)
+
+        self.log.info(self.connection.send_message(**params))
+
+
+__all__ = [TelegramHook]
```

Comment:
```suggestion
__all__ = ["TelegramHook"]
```
This should be a list of module names, not modules.

However, I would like to point out that we do not use the __all__ variable in the whole project. I think that this is a good rule because it forces you to think about the whole project, not just small fragments. In the future, when Airflow will be divided into modules and interfaces between libraries will be important, we can think about changing this rule. Now we have one code base and we must think about its overall good.

Why is the decorator only in this module, not in the entire Airflow?  What if another operator needs this decorator? Copy code? Move code?  If we copy to another place, what will we do with the documentation? Can we change it? Did we respect code style of author?

@zhongjiajie What are you thinking?